### PR TITLE
Add supoort for Tls 1.3 in Web cmdlets

### LINKED ISSUE #13398: Support for TLS 1.3
## Steps to reproduce
[enum]::GetValues('Net.SecurityProtocolType')
Invoke-WebRequest -SslProtocol tls13 -Uri https://www.google.com

image

## Expected behavior
Connect with TLS 1.3 without error

## Actual behavior
Error message:
Invoke-WebRequest: Cannot bind parameter 'SslProtocol'. Cannot convert value "tls13" to type "Microsoft.PowerShell.Commands.WebSslProtocol". Error: "Unable to match the identifier name tls13 to a valid enumerator name. Specify one of the following enumerator names and try again:
Default, Tls, Tls11, Tls12"

## Environment data

Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```none

```

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #13398 

Add new element in WebSslProtocol enum.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/6481
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by vexx32 — APPROVED** [review]

**Review by rjmholt — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
Would be great if we can add a test.

**Comment by iSazonov** [comment]
I added tests.

**Comment by iSazonov** [comment]
After investigations of new test failures I found that .Net delegates the TLS support to underlying OS. The TLS 1.3 support on all OS-s is limited.
I enable TLS 1.3 on Windows 10 1903 and get:
```powershell
Invoke-WebRequest -SslProtocol tls13 https://tls13.akamai.io/

Invoke-WebRequest: Authentication failed because the remote party sent a TLS alert: 'HandshakeFailure'.
```
MacOS haven't the support at all. Linux - depends on used OpenSSL lib.


@SteveL-MSFT We need a conclusion:
- remove all new tests and our test WebListener updates
- put new tests in pending
- ???

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 1]
```diff
@@ -1694,26 +1694,36 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
...
                 @{ Test = @{SslProtocol = 'Tls, Tls11'; ActualProtocol = 'Tls11'}; Pending = $false }
                 @{ Test = @{SslProtocol = 'Tls, Tls11, Tls12'; ActualProtocol = 'Tls'}; Pending = $false }
+                @{ Test = @{SslProtocol = 'Tls, Tls11, Tls13'; ActualProtocol = 'Tls'}; Pending = $false }
                 @{ Test = @{SslProtocol = 'Tls, Tls11'; ActualProtocol = 'Tls'}; Pending = $false }
                 # Skipping intermediary protocols is not supported on all platforms
                 @{ Test = @{SslProtocol = 'Tls, Tls12'; ActualProtocol = 'Tls'}; Pending = -not $IsWindows }
                 @{ Test = @{SslProtocol = 'Tls, Tls12'; ActualProtocol = 'Tls12'}; Pending = -not $IsWindows }
             )
 
             $testCases2 = @(
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:1713]: Can you add a comment that these are failing test cases

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 2]
```diff
@@ -3196,26 +3206,36 @@ Describe "Invoke-RestMethod tests" -Tags "Feature", "RequireAdminOnWindows" {
...
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:3225]: similar comment here

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
> @SteveL-MSFT We need a conclusion:
> - remove all new tests and our test WebListener updates
> - put new tests in pending
> - ???

Put the tests at pending and file an issue.  Note the issue in a comment in the test code.

Best if you pend the test cases like this:
https://github.com/PowerShell/PowerShell/blob/f4382202ae4622bf26795e29a7b39b9d7cdfb3fb/test/powershell/Modules/PSDesiredStateConfiguration/PSDesiredStateConfiguration.Tests.ps1#L129

**Comment by iSazonov** [comment]
@TravisEz13 Tracking issue is created and the comment is added to the tests. (I can not use `Set-ItResult` without refactoring the tests.)

**Comment by iSazonov** [comment]
Rebased to pass CIs.

**Comment by rjmholt** [comment]
@iSazonov looks like this may need another rebase. Once CI passes, I'll merge

**Comment by rjmholt** [comment]
CI still failing

**Pull request merged**